static/absolute/js/searchPage.js

```diff
@@ -110,45 +110,54 @@ socket.on("refresh", () => {
 
 async function refreshUI() {
   const eventId = getEventId();
-  const participantsResponse = await (
-    await fetch(`/api/${eventId}/participants`)
-  ).json();
-  const restaurantsResponse = await (
-    await fetch(`/api/${eventId}/restaurants`)
-  ).json();
 
-  initMap(participantsResponse, restaurantsResponse);
-  showRestaurants(restaurantsResponse);
+  // Fetch in parallel
+  const [participantsResponse, restaurantsResponse] = await Promise.all([
```

Comment:
this blocks until both fetches have completed successfully - however we can procede with showRestaurants as soon as one is complete. can you refactor to allow for that? for extra credit, initMap could perhaps also be split into two functions, one for participants, one for restaurants, to allow for further latency improvements?

Reply:
Done